drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -68,14 +85,135 @@ static uint32_t overrides_sub_ghz[] = {
+	.symbolRate.preScale = 15,
+	.symbolRate.rateWord = 131072,
+	.rxBw = 0x59,                   /* 310.8 kHz */
+	.preamConf.nPreamBytes = 7,
+	.formatConf.nSwBits = 24,       /* 24-bit of syncword */
+	.formatConf.bMsbFirst = true,
+	.formatConf.whitenMode = 7,
+	.config.biasMode = true,
+	.formatConf.bMsbFirst = true,
+	.txPower = 0x013f, /* from Smart RF Studio */
+	.centerFreq = 915,
+	.intFreq = 0x0999,
+	.loDivider = 5,
+	.pRegOverride = ieee802154_cc13xx_overrides_sub_ghz,
 };
 
+/* Radio values for CC13X2P */
+#elif defined(CONFIG_SOC_CC1352P)
+/* CC1352P overrides from SmartRF Studio (200kbps, 50kHz deviation, 2-GFSK, 311.8kHz Rx BW) */
+static uint32_t ieee802154_cc13xx_overrides_sub_ghz[] = {
+	/* Tx: Configure PA ramp time, PACTL2.RC=0x3 (in ADI0, set PACTL2[4:3]=0x1) */
+	ADI_2HALFREG_OVERRIDE(0, 16, 0x8, 0x8, 17, 0x1, 0x0),
+	/* Rx: Set AGC reference level to 0x16 (default: 0x2E) */
+	HW_REG_OVERRIDE(0x609C, 0x0016),
+	/* Rx: Set RSSI offset to adjust reported RSSI by -1 dB (default: -2), trimmed */
+	/* for external bias and differential configuration */
+	(uint32_t)0x000188A3,
+	/* Rx: Set anti-aliasing filter bandwidth to 0x6 (in ADI0, set IFAMPCTL3[7:4]=0x8) */
+	ADI_HALFREG_OVERRIDE(0, 61, 0xF, 0x8),
+	/* override_prop_common_sub1g.xml */
+	/* Set RF_FSCA.ANADIV.DIV_SEL_BIAS = 1. Bits [0:16, 24, 30] are don't care.. */
+	(uint32_t)0x4001405D,
+	/* Set RF_FSCA.ANADIV.DIV_SEL_BIAS = 1. Bits [0:16, 24, 30] are don't care.. */
+	(uint32_t)0x08141131,
+	/* override_prop_common.xml */
+	/* DC/DC regulator: In Tx with 14 dBm PA setting, use DCDCCTL5[3:0]=0xF */
+	/* (DITHER_EN=1 and IPEAK=7). In Rx, use default settings. */
+	(uint32_t)0x00F788D3,
+	/* TX power override */
+	/* Tx: Set PA trim to max to maximize its output power (in ADI0, set PACTL0=0xF8) */
+	ADI_REG_OVERRIDE(0, 12, 0xF8),
+	(uint32_t)0xFFFFFFFF
+};
+static uint32_t rf_prop_overrides_tx_std[] = {
+	/* The TX Power element should always be the first in the list */
+	TX_STD_POWER_OVERRIDE(0xB224),
+	/* The ANADIV radio parameter based on the LO divider (0) and front-end (0) settings */
+	(uint32_t)0x11310703,
+	/* override_phy_tx_pa_ramp_genfsk_std.xml */
+	/* Tx: Configure PA ramping, set wait time before turning off */
+	/* (0x1A ticks of 16/24 us = 17.3 us). */
+	HW_REG_OVERRIDE(0x6028, 0x001A),
+	(uint32_t)0xFFFFFFFF
+};
+static uint32_t rf_prop_overrides_tx_20[] = {
+	/* The TX Power element should always be the first in the list */
+	TX20_POWER_OVERRIDE(0x001B8ED2),
+	/* The ANADIV radio parameter based on the LO divider (0) and front-end (0) settings */
+	(uint32_t)0x11C10703,
+	/* override_phy_tx_pa_ramp_genfsk_hpa.xml */
+	/* Tx: Configure PA ramping, set wait time before turning off */
+	/* (0x1F ticks of 16/24 us = 20.3 us). */
+	HW_REG_OVERRIDE(0x6028, 0x001F),
+	(uint32_t)0xFFFFFFFF
+};
+
+/* Radio setup command for CC1312P / CC1352P */
+static volatile rfc_CMD_PROP_RADIO_DIV_SETUP_PA_t ieee802154_cc13xx_subg_radio_div_setup = {
```

Comment:
My use case here was for an application to configure custom / proprietary RF parameters (e.g., different bit rates / modulations for better signal resiliency, as generated by SmartRF Studio).
However, a Device Tree Binding for this needs to model all of the RF properties (and there are dozens). But I guess it would still be more elegant than our current approach.